src/app/launcher/create-app/dependency-editor-step/dependency-editor-step.component.ts

```diff
@@ -0,0 +1,211 @@
+                this.listenForChanges(r);
+            });
+        }
+    }
+
+    // Accessors
+    /**
+     * Returns indicator that step is completed
+     *
+     * @returns {boolean} True if step is completed
+     */
+    get stepCompleted(): boolean {
+        return (this.launcherComponent.summary.dependencyEditor !== undefined);
+    }
+
+    /**
+     * Returns target environments to display
+     *
+     * @returns {TargetEnvironment[]} The target environments to display
+     */
+    // get targetEnvironments(): TargetEnvironment[] {
+    //   return this._targetEnvironments;
+    // }
+    // Steps
+    navToNextStep(): void {
+        this.launcherComponent.navToNextStep();
+    }
+
+    /**
+     * Navigate to step
+     *
+     * @param {string} id The step ID
+     */
+    navToStep(event: string) {
+        this.launcherComponent.stepIndicator.navToStep(event);
+    }
+
+    updateTargetEnvSelection(): void {
+        this.initCompleted();
+    }
+
+    public pickDependencies(event: any) {
+        if (event) {
+            this.launcherComponent.summary.dependencyEditor['dependencySnapshot'] = event;
+        }
+    }
+
+    public pickMetadata(event: any) {
+        if (event) {
+            this.launcherComponent.summary.dependencyCheck.mavenArtifact = event.artifactId;
+            this.launcherComponent.summary.dependencyCheck.groupId = event.groupId;
+            this.launcherComponent.summary.dependencyCheck.projectVersion = event.version;
+
+            // Update the dependency editor model
+            this.launcherComponent.summary.dependencyEditor.mavenArtifact = event.artifactId;
+            this.launcherComponent.summary.dependencyEditor.groupId = event.groupId;
+            this.launcherComponent.summary.dependencyEditor.projectVersion = event.version;
+        }
+        setTimeout(() => {
+            this.initCompleted();
+        }, 10);
+    }
+
+    // Private
+    private initCompleted(): void {
+        this.launcherComponent.getStep(this.id).completed = this.stepCompleted;
+    }
+
+    private listenForChanges(change: any): void {
+        let flag = false;
+        if (change && change.key === 'runtime') {
+            let current = change.currentValue;
+
+            if (
+                !this.cacheInfo ||
+                (
+                    this.cacheInfo && this.cacheInfo.id !== current.id
+                )
+            ) {
+                // Changes in any of them: name or version.id or version.name
+                this.cacheInfo['runtime'] = {
+                    name: current.name,
+                    id: current.id,
+                    version: current.version.id
+                };
+                flag = true;
+            }
+        } else if (change && change.key === 'mission') {
+            let current = change.currentValue;
+            this.cacheInfo['mission'] = {
+                id: current.id
+            };
+            flag = true;
+        }
+
+        if (flag) {
```

Comment:
Can this logic be moved to separate method? WDYT

Reply:
Yes it can be but this logic is not used anywhere else separately, so I think it'll be better to use it with the existing method